File: sortAlgorithim/Java/QuickSort.java
package sortAlgorithm;

public class QuickSort {
	public static void quickSort(int[]arr,int start,int end) {
		if(start<end) {
			int index = getIndex(arr,start,end);
			quickSort(arr,start,index-1);
			quickSort(arr,index+1,end);
		}
	}
	public static int getIndex(int[]arr,int start,int end) {
		int tmp = arr[start];
		while(start<end){
			while(start<end && tmp<=arr[end]) end--;
			arr[start] = arr[end];
			while(start<end && tmp>=arr[start]) start++;
			arr[end] = arr[start];
		}
		arr[start] = tmp;
		return start;
	}
	public static void main(String[] args) {
		int[] a = {5,4,15,2,1,6,20,3,7,5,14};
		System.out.print("排序前数组a：\n");
		for(int i:a) {
			System.out.print(i);
			System.out.print(" ");	
		}
		quickSort(a,0,a.length-1);
		System.out.print("\n排序后数组a：\n");
		for(int i:a) {
			System.out.print(i);
			System.out.print(" ");
		}
	}
}

File: sortAlgorithim/Go/QuickSort.go
package main

import "fmt"

func show(nums []int){
	for _,num:=range nums{
		fmt.Printf("%d ",num)
	}
	fmt.Println()
}

func getIndex(nums []int,start int,end int)(int,[]int){
	tmp:= nums[start]
	for ;start<end;{
		for;start<end&&tmp<=nums[end];{
			end--
		}
		nums[start] = nums[end]
		for;start<end&&tmp>=nums[start];{
			start++
		}
		nums[end] = nums[start]
	}
	nums[start] = tmp
	return start,nums
}

func quickSort(nums []int,start int,end int)[]int{
	if(start<end){
		index,nums:= getIndex(nums,start,end)
		nums = quickSort(nums,0,index-1)
		nums = quickSort(nums,index+1,end)
	}
	return nums
}

func main(){
	nums :=[]int{5,6,3,1,8,7,2,4}
	fmt.Println("排序前：")
	show(nums)
	quickSort(nums,0,len(nums)-1)
	fmt.Println("排序后：")
	show(nums)
}
